head/head.go

```diff
@@ -83,10 +82,19 @@ func NewHead(ctx context.Context, options ...opts.Option) (*Head, chan Bootstrap
 		return nil, nil, fmt.Errorf("failed to spawn libp2p node: %w", err)
```

Comment:
Ideally, we'd also pass `libp2p.EnableNATService()` and `libp2p.AutoNATServiceRateLimit(0, 3, time.Minute)`  in the libp2p options.